app/services/sandbox_seed_data/school_partnerships.rb

```diff
@@ -15,14 +15,30 @@ def plant
   private
 
     def create_school_partnerships
-      LeadProviderDeliveryPartnership.find_each do |lead_provider_delivery_partnership|
-        school = School.order("RANDOM()").first
-
-        next if school.school_partnerships.map(&:lead_provider_delivery_partnership).map(&:id).include?(lead_provider_delivery_partnership.id)
+      ids = LeadProviderDeliveryPartnership.pluck(:id)
+      LeadProviderDeliveryPartnership.where(id: ids).find_each do |lead_provider_delivery_partnership|
+        existing_school_ids = lead_provider_delivery_partnership.school_partnerships.pluck(:school_id)
+        school = School.where.not(id: existing_school_ids).order("RANDOM()").first
+        next unless school
 
         school_partnership = FactoryBot.create(:school_partnership, school:, lead_provider_delivery_partnership:)
 
         log_seed_info("#{school_partnership.school.name} -> #{school_partnership.lead_provider.name}", colour: Colourize::COLOURS.keys.sample)
+
+        # Create another school partnership with the same school but different delivery_partner
+        if rand < 0.1
```

Comment:
instead of relying on randomness can we ensure there are at least 2 school partnerships for lp/contract period/school with different dps?

Reply:
Refactored to allow specifying number of records